test_install/CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ if((NOT GINKGO_BUILD_SHARED_LIBS) AND GINKGO_BUILD_HIP)
     endif()
 endif()
 
+if(GINKGO_BUILD_OMP)
+  find_package(OpenMP REQUIRED)
+endif()
```

Comment:
Do you really need to use the `OpenMP::OpenMP_CXX` target? I would rather we don't have an extra public OpenMP dependency in this way.

Reply:
I could make [this linking](https://github.com/ginkgo-project/ginkgo/blob/083a5dee7a81aea745d02c9d58f81a6dec373f5b/omp/CMakeLists.txt#L30) `PUBLIC`, but I guess that is also a public dependency ?

Reply:
Well yes, that would make it spill over everywhere just like now. What I don't get is, is `OpenMP_CXX_LIBRARIES` wrong on MacOS? Or empty? What is the issue? I believe it should be available everywhere since it's a CMake standard from FindOpenMP.

Reply:
One minor point: We don't set the OpenMP include paths at the moment

Reply:
In general with OpenMP that should not be needed, because OpenMP is integrated into the compilers and they take care of it. The OpenMP header usually sits in the compiler's private directory (you can activate it with flags, such as `-fopenmp` or whichever is your compiler's flag).

Reply:
AppleClang is a bit... difficult in that regard ;) It also seems like the detected flag is incorrect: `clang: warning: argument unused during compilation: '-Xclang -fopenmp' [-Wunused-command-line-argument]`
[This article](https://iscinumpy.gitlab.io/post/omp-on-high-sierra/) suggests we should actually be using `-Xpreprocessor -fopenmp`

Reply:
@upsj, looks like that minor point was the issue. It looks like it works now. :+1:

Reply:
Good find @upsj. I guess FindOpenMP's flags are for AppleClang then. They also don't seem to want to fix it:
https://github.com/Kitware/CMake/blob/b0e34a3160642a09728f1f56d93e2835443eb016/Modules/FindOpenMP.cmake#L100

Reply:
I definitely remember `-Xclang -fopenmp` working with older versions of XCode/AppleClang, but this is something that should be solved upstream in CMake.